src/core/hle/libraries/libkernel/libkernel.cpp

```diff
@@ -86,6 +89,14 @@ int PS4_SYSV_ABI sceKernelMmap(void* addr, u64 len, int prot, int flags, int fd,
     }
     *res = ret;
     return 0;
+#else
+    void* result = mmap(addr, len, prot, flags, fd, offset);
+    if (result != MAP_FAILED) {
+        *res = result;
+        return 0;
+    }
+    std::abort();
```

Comment:
This will probably compile on all major compilers still, but it's UB to have a path that doesn't return a value even if you are guaranteed to abort with `std::abort()`